returnn/tf/layers/basic.py

```diff
@@ -913,37 +913,204 @@ def transform_config_dict(cls, d, network, get_layer):
 
 class GatherLayer(_ConcatInputLayer):
   """
-  This is basically ``x[position]`` in some axis.
-  See also :class:`SliceLayer`.
-  See also :class:`GatherNdLayer`, which works on the time-axis, but supports different positions per batch.
+  Gathers slices on a specified axis from the input layer using indices from a ``position`` layer.
+  If the input is a layer of the shape ``[B,D,F1]``, and indices of shape ``[B,F2]``, this will yield output of the
+  shape ``[B,F2,F1]`` where
 
-  (Currently `position` can just be an integer, so it's much less generic than ``tf.gather``.
-   However, we might extend this to lists, arrays, or even other layers.)
+  ``output[b,f2,f1] = input[b,indices[b,f2],f1]``
```

Comment:
I would also explain the simple case, i.e. just give the example `input[position]`.

And extend it to explain the difference to `GatherNdLayer` (can be short, just mention this is a more generic version of `GatherNdLayer`).

And extend the `GatherNdLayer` documentation to say it is deprecated and `GatherLayer` should be used instead.